<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>961</width>
    <height>525</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>525</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Lab 3 - Phong lighting</string>
  </property>
  <widget class="QWidget" name="centralWidget"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>961</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QDockWidget" name="settingsDock">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>295</width>
     <height>580</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetVerticalTitleBar</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::NoDockWidgetArea</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="settingsDockContent">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>275</width>
      <height>0</height>
     </size>
    </property>
    <property name="contextMenuPolicy">
     <enum>Qt::DefaultContextMenu</enum>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <layout class="QVBoxLayout" name="settingsLayout">
     <item>
      <widget class="QGroupBox" name="lightGroupBox">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>105</height>
        </size>
       </property>
       <property name="title">
        <string>Light</string>
       </property>
       <widget class="QSlider" name="lightIntensitySlider">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>70</y>
          <width>101</width>
          <height>22</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="lightIntensityLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>70</y>
          <width>91</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Intensity</string>
        </property>
       </widget>
       <widget class="QPushButton" name="lightColorButton">
        <property name="geometry">
         <rect>
          <x>80</x>
          <y>30</y>
          <width>41</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="lightColorLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>62</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Color</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="lightIntensityEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>70</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="lightREdit">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>30</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="lightGEdit">
        <property name="geometry">
         <rect>
          <x>180</x>
          <y>30</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="lightBEdit">
        <property name="geometry">
         <rect>
          <x>220</x>
          <y>30</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="attenuationGroupBox">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>105</height>
        </size>
       </property>
       <property name="title">
        <string>Light Attenuation</string>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
       <widget class="QSlider" name="attQuadraticSlider">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>20</y>
          <width>101</width>
          <height>22</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="attLinearLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>61</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Linear</string>
        </property>
       </widget>
       <widget class="QLabel" name="attConstantLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>70</y>
          <width>91</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Constant</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="attConstantEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>80</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QSlider" name="attLinearSlider">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>50</y>
          <width>101</width>
          <height>22</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLineEdit" name="attLinearEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>50</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QSlider" name="attConstantSlider">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>80</y>
          <width>101</width>
          <height>22</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLineEdit" name="attQuadraticEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>20</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="attQuadraticLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>91</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Quadratic</string>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="spheresGroupBox">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>115</height>
        </size>
       </property>
       <property name="title">
        <string>Sphere Colors</string>
       </property>
       <widget class="QPushButton" name="sphereLColorButton">
        <property name="geometry">
         <rect>
          <x>80</x>
          <y>30</y>
          <width>41</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="sphereLLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>62</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Left</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="sphereLREdit">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>30</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="sphereLGEdit">
        <property name="geometry">
         <rect>
          <x>170</x>
          <y>30</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="sphereLBEdit">
        <property name="geometry">
         <rect>
          <x>210</x>
          <y>30</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="sphereMBEdit">
        <property name="geometry">
         <rect>
          <x>210</x>
          <y>60</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="sphereMLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>50</y>
          <width>62</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Middle</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="sphereMGEdit">
        <property name="geometry">
         <rect>
          <x>170</x>
          <y>60</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="sphereMREdit">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>60</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="sphereMColorButton">
        <property name="geometry">
         <rect>
          <x>80</x>
          <y>60</y>
          <width>41</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QPushButton" name="sphereRColorButton">
        <property name="geometry">
         <rect>
          <x>80</x>
          <y>90</y>
          <width>41</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="sphereRLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>80</y>
          <width>62</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Right</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="sphereRGEdit">
        <property name="geometry">
         <rect>
          <x>170</x>
          <y>90</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="sphereRBEdit">
        <property name="geometry">
         <rect>
          <x>210</x>
          <y>90</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="sphereRREdit">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>90</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="materialGroupBox">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>219</height>
        </size>
       </property>
       <property name="title">
        <string>Sphere Material</string>
       </property>
       <widget class="QLabel" name="ambientLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>71</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Ambient</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="ambientEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>30</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QSlider" name="ambientSlider">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>30</y>
          <width>101</width>
          <height>22</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="diffuseLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>50</y>
          <width>71</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Diffuse</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="diffuseEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>60</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QSlider" name="diffuseSlider">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>60</y>
          <width>101</width>
          <height>22</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="specularLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>80</y>
          <width>81</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Specular</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="specularEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>90</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QSlider" name="specularSlider">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>90</y>
          <width>101</width>
          <height>22</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="shininessLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>110</y>
          <width>81</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Shininess</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="shininessEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>120</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QSlider" name="shininessSlider">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>120</y>
          <width>101</width>
          <height>22</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <designerdata>
  <property name="gridDeltaX">
   <number>10</number>
  </property>
  <property name="gridDeltaY">
   <number>10</number>
  </property>
  <property name="gridSnapX">
   <bool>true</bool>
  </property>
  <property name="gridSnapY">
   <bool>true</bool>
  </property>
  <property name="gridVisible">
   <bool>true</bool>
  </property>
 </designerdata>
</ui>
